commit 785d94ee789f4ec706e19832e24e1298d263fa29
Author: Jon Maxwell <jmaxwell37@gmail.com>
Date:   Fri May 25 07:38:29 2018 +1000

    ifb: fix packets checksum
    
    commit b1d2e4e03f92734ff524f96c4b2287133de7a4a3 upstream.
    
    Fixup the checksum for CHECKSUM_COMPLETE when pulling skbs on RX path.
    Otherwise we get splats when tc mirred is used to redirect packets to ifb.
    
    Before fix:
    
    nic: hw csum failure
    
    Signed-off-by: Jon Maxwell <jmaxwell37@gmail.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/drivers/net/ifb.c b/drivers/net/ifb.c
index 5f2897ec0edc..d345c61d476c 100644
--- a/drivers/net/ifb.c
+++ b/drivers/net/ifb.c
@@ -102,7 +102,7 @@ static void ifb_ri_tasklet(unsigned long _txp)
 		if (!skb->tc_from_ingress) {
 			dev_queue_xmit(skb);
 		} else {
-			skb_pull(skb, skb->mac_len);
+			skb_pull_rcsum(skb, skb->mac_len);
 			netif_receive_skb(skb);
 		}
 	}
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1197_785d94ee789f4ec706e19832e24e1298d263fa29.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1197_785d94ee789f4ec706e19832e24e1298d263fa29.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
